runtime/src/runtime/fvm.rs

```diff
@@ -443,6 +443,14 @@ where
         fvm::crypto::recover_secp_public_key(hash, signature)
             .map_err(|e| anyhow!("failed to recover pubkey; exit code: {}", e))
     }
+
+    fn verify_replica_update(&self, replica: &ReplicaUpdateInfo) -> Result<(), Error> {
+        match fvm::crypto::verify_replica_update(replica) {
+            Ok(true) => Ok(()),
+            Ok(false) => Err(Error::msg("invalid replica")),
+            Err(e) => Err(anyhow!("failed to verify replica: {}", e)),
+        }
+    }
```

Comment:
I don't understand what this change is achieving, nor the similar one in MockRuntime. These two don't need injectable primitives.

Reply:
the intent here is not to make the behaviour injectable rather to have a consistent location for implementation. 

the Runtime implements both Verifier and Primitives. The idea of this design is that all behaviour described by the Verifier will be duplicated in Primitives. Verifier should delegate its calls to the Primitives implementation (in the concrete FVM case, it would also work to duplicate the code in both trait impls but I thought this pattern would be more consistent).

By having Verifier delegate to Primitives (in the TestVM) it allows all Verifier behaviour to be modifiable since FakePrimitives can have mock behaviour injected.